libs/transloco-locale/src/lib/pipes/transloco-decimal.pipe.ts

```diff
@@ -31,12 +31,39 @@ export class TranslocoDecimalPipe
    * 1234567890 | translocoDecimal: {useGrouping: false}: en-US // 1234567890
    *
    */
+  // overloads for strict mode
   transform(
-    value: string | number,
+    value: number | string,
+    numberFormatOptions?: NumberFormatOptions,
+    locale?: Locale
+  ): string;
+  transform(
+    value: null | undefined,
+    numberFormatOptions?: NumberFormatOptions,
+    locale?: Locale
+  ): null | undefined;
+  transform(
+    value: number | string | null,
+    numberFormatOptions?: NumberFormatOptions,
+    locale?: Locale
+  ): string | null;
+  transform(
+    value: number | string | undefined,
+    numberFormatOptions?: NumberFormatOptions,
+    locale?: Locale
+  ): string | undefined;
+  transform(
+    value: number | string | null | undefined,
+    numberFormatOptions?: NumberFormatOptions,
+    locale?: Locale
+  ): string | null | undefined;
+
+  transform(
+    value?: string | number | null,
```

Comment:
The overloading seems robust.
Why not simply:
```ts
  transform(
      value: string | number,
      numberFormatOptions?: NumberFormatOptions,
      locale?: Locale
  ): string;
  transform<T extends undefined | null>(
      value: T,
      numberFormatOptions?: NumberFormatOptions,
      locale?: Locale
  ): T;
  transform(
    value: string | number | undefined | null,
    numberFormatOptions: NumberFormatOptions = {},
    locale?: Locale
  )
```
Seems like to cover the cases corretly

Reply:
Don't know why I didn't think of this. I adjusted the commits, thanks.